### ISSUE TITLE
Re-enable flaky SPT tests for least upper bound on datatypes

### ISSUE THREAD

**Issue by MeAmAnUsername** [MeAmAnUsername/pie#99]
There are three SPT tests for least upper bound on datatypes (`expression.spt` --> `ifelse`) that sometimes fail and sometimes pass.
This has been reported in the `spoofax-users` channel on Slack: [thread here](https://slde.slack.com/archives/C7254SF60/p1620998582250500)
It seems to be a bug in Statix.
The tests are currently commented out with a todo referencing this issue, they should be enabled again when the Statix bug is fixed.

**Comment by MeAmAnUsername** [comment]
Gabriel has updated the Spoofax Gradle plugin, so the tests should pass reliably again.

**Comment by MeAmAnUsername** [comment]
Tests are reliable again, unfortunately, some of them fail:
```
//    generic1: Generic[Foo, Baar, string],
//    genericSub: GenericSub[Bar, Baar]
val res1: Generic[_ : Foo, _ : Baar, _] = if (value == 0) genericSub else generic1;
val res2: Generic[_ : Foo, Baar, _] = if (value == 0) genericSub else generic1;
```
`res1` works, but it should be like `res2`, where it uses `Baar` instead of `_ : Baar`

**Comment by MeAmAnUsername** [comment]
Bug has been solved, tests are now enabled again. Will be merged with generics, see #24.
